presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -975,9 +976,21 @@ protected Object visitLikePredicate(LikePredicate node, Object context)
             if (pattern instanceof Slice && escape == null) {
                 String stringPattern = ((Slice) pattern).toStringUtf8();
                 if (!stringPattern.contains("%") && !stringPattern.contains("_")) {
-                    return new ComparisonExpression(ComparisonExpression.Type.EQUAL,
-                            toExpression(value, expressionTypes.get(node.getValue())),
-                            toExpression(pattern, expressionTypes.get(node.getPattern())));
+                    Type valueType = expressionTypes.get(node.getValue());
```

Comment:
I was wondering whether it's a good idea to make assumption that equals always take two values of the same type. And then I realize we are not making such an assumption in general because we know both side are `VARCHAR` or `VARCHAR(x)`. Do you think it's fair to assert that both sides are `VARCHAR` or `VARCHAR(x)` here?